tests/test_add_fund_and_claim.cairo

```diff
@@ -0,0 +1,418 @@
+   
+
+    start_prank(usdc, deployer_addr());
+    usdc_dispatcher.approve(salary_address, 300 * TOKEN_MULTIPLIER);
+    stop_prank(usdc);
+
+    start_prank(salary_address, deployer_addr());
+    salary_dispatcher.add_fund_to_salary_pools(092023, amounts1, guilds.clone());
+    salary_dispatcher.add_fund_to_salary_pools(102023, amounts2, guilds.clone());
+    stop_prank(salary_address);
+
+
+    // verifying balance of contract 
+    let contract_balance = usdc_dispatcher.balance_of(salary_address);
+    assert(contract_balance == 300 * TOKEN_MULTIPLIER, 'incorrect balance');
+
+    let user1_expected_cum_salary = (120 * 10 * TOKEN_MULTIPLIER / 320) + (150 * 50 * TOKEN_MULTIPLIER / 250) + 
+                                    (250 * 20 * TOKEN_MULTIPLIER / 400) + (20 * 40 * TOKEN_MULTIPLIER / 20) +
+                                    (30 * 30 * TOKEN_MULTIPLIER / 30) + (45 * 30 * TOKEN_MULTIPLIER / 100) +
+                                    (20 * 40 * TOKEN_MULTIPLIER / 120) + (0 * 20 * TOKEN_MULTIPLIER / 50) +
+                                    (10 * 50 * TOKEN_MULTIPLIER / 80) + (35 * 10 * TOKEN_MULTIPLIER / 125);
+
+    let user2_expected_cum_salary = (200 * 10 * TOKEN_MULTIPLIER / 320) + (100 * 50 * TOKEN_MULTIPLIER / 250) + 
+                                    (150 * 20 * TOKEN_MULTIPLIER / 400) + (0 * 40 * TOKEN_MULTIPLIER / 20) +
+                                    (0 * 30 * TOKEN_MULTIPLIER / 30) + (55 * 30 * TOKEN_MULTIPLIER / 100) +
+                                    (100 * 40 * TOKEN_MULTIPLIER / 120) + (50 * 20 * TOKEN_MULTIPLIER / 50) +
+                                    (70 * 50 * TOKEN_MULTIPLIER / 80) + (90 * 10 * TOKEN_MULTIPLIER / 125);
+    
+    
+    let user1_cum_salary = salary_dispatcher.get_cum_salary(user1());
+    assert(user1_cum_salary == user1_expected_cum_salary,'incorrect user1 cum salary');
+
+    let user2_cum_salary = salary_dispatcher.get_cum_salary(user2());
+    assert(user2_cum_salary == user2_expected_cum_salary,'incorrect user2 cum salary');
+
+    let mut spy = spy_events(SpyOn::One(salary_address));
+    // claiming the salary
+    start_prank(salary_address, user1());
+    salary_dispatcher.claim_salary(user1());
+    stop_prank(salary_address);
+
+    let mut event_data1 = Default::default();
+    Serde::serialize(@user1_expected_cum_salary, ref event_data1);
+    Serde::serialize(@user1(), ref event_data1);
+    spy.assert_emitted(@array![
+        Event { from: salary_address, name: 'SalaryClaimed', keys: array![], data: event_data1 }
+    ]);
+
+
+    // verfying tokens transfered successful
+    let user1_balance = usdc_dispatcher.balance_of(user1());
+    assert(user1_balance == user1_expected_cum_salary, 'incorrect user1 balance');
+
+    // verifying claimed_salary is updated
+    let user1_claimed_salary = salary_dispatcher.get_claimed_salary(user1());
+    assert(user1_claimed_salary == user1_expected_cum_salary, 'incorrect claimed amount');
```

Comment:
Could you also check if the balance of the salary contract has changed? As you did in the line 343-344

Reply:
done